src/simple.ts

```diff
@@ -1,51 +1,54 @@
 
-import Promise = require("q");
+import Promise = require("bluebird");
 
-import { Locator, Location } from "./common";
+import {Locator, Location, LocatorFactory} from "./common";
 
-var integerRegExp = /^\d+$/;
+const integerRegExp = /^\d+$/;
 
 export interface SimpleLocatorParameters {
   resource : string;
   defaultPort? : number;
 }
 
-export function simpleLocatorFactory() : Function {
-  function simpleLocator(parameters : any) : Locator {
-    if (typeof parameters === "string") {
-      parameters = { resource : parameters };
+export class SimpleLocator {
+    static getLocatorFactory() : LocatorFactory {
+        return function simpleLocator(parameters : string|SimpleLocatorParameters) : Locator {
```

Comment:
i purposefully did these as named functions to perhaps help with stack traces and the like, since arrow functions are anonymous. They are static and as such are not calling `this`, so arrow functions don't really add anything in this case.